bpf/lib/nodeport_egress.h

```diff
@@ -306,13 +306,34 @@ static __always_inline int nodeport_snat_fwd_ipv4(struct __ctx_buff *ctx,
 	void *data, *data_end;
 	struct iphdr *ip4;
 	int l4_off, ret;
+	struct endpoint_info *ep;
 
 	if (!revalidate_data(ctx, &data, &data_end, &ip4))
 		return DROP_INVALID;
 
 	snat_v4_init_tuple(ip4, NAT_DIR_EGRESS, &tuple);
 	l4_off = ETH_HLEN + ipv4_hdrlen(ip4);
 
+	ep = __lookup_ip4_endpoint(ip4->saddr);
+	if (ep && ep->parent_ifindex && ep->parent_ifindex != NATIVE_DEV_IFINDEX) {
+		/* This packet came from an endpoint with a parent interface and
+		 * it us currently not egressing on its parent interface.
+		 * Check if its a reply packet, if it is, redirect it to the
+		 * parent interface.
+		 */
+		if (l4_load_ports(ctx, l4_off, (__be16 *)&tuple.dport) < 0)
+			return DROP_INVALID;
+
```

Comment:
To handle fragmentation, this would need to be `ipv4_load_l4_ports()`. Essentially what happens in `ct_extract_ports4()` when `snat_v4_needs_masquerade()` calls `ct_is_reply4()`.